### ISSUE TITLE
What's the purpose of combining OCP and ASPP/PPM

### ISSUE THREAD

**Issue by yaoqi-zd** [PkuRainBow/OCNet.pytorch#37]
Hi, May I ask you why do you combine aspp/ppm with ocp, since you have claimed aspp/ppm would introduce information from other categories to the target pixel and make the prediction unreliable, the reasonable design seems to replace aspp/ppm with ocp rather than combine with them?

**Comment by PkuRainBow** [comment]
Here we mainly refers to the global average pooling operation within the ASPP/PPM, thus we replace all the global average pooling with OCP.

**Comment by yaoqi-zd** [comment]
Thanks for your reply! in my sense, the ocp module models the pixel similarities at all distances and thus capturing both short and long region dependencies. could you explain why the rest dilated convolution in ASPP module can still contribute to the performance (base-oc:78.80 vs asp-oc:79.58) ?

**Comment by PkuRainBow** [comment]
In fact, the base-oc can achieve more than 79.3+ on the validation set of the Cityscapes according to our current experiments. But asp-oc performs more reliable on the test set. We guess that the self-attention method is good but not generalize well as ASPP.

**Comment by yaoqi-zd** [comment]
I get it! Thanks for your valuable discussion
